diff --git a/dist/index.d.ts b/dist/index.d.ts
index 590773839c4fb595e5284247ed3cf75a6624e3c3..5f174e4c6bf8d6bf6baf15d2f8ffaba9b0869731 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -6326,8 +6326,99 @@ interface VitestPluginSettings {
 //#region src-js/package/config.d.ts
 type Oxlintrc = Omit<Oxlintrc$1, "$schema" | "extends">;
 type ExternalPluginsConfig = Exclude<Oxlintrc["jsPlugins"], undefined | null>;
-interface OxlintConfig extends Oxlintrc {
+interface OxlintConfig {
+  /**
+   * Configure an entire category of rules all at once.
+   *
+   * Rules enabled or disabled this way will be overwritten by individual rules in the `rules` field.
+   *
+   * Example
+   * ```json
+   * {
+   *   "$schema": "./node_modules/oxlint/configuration_schema.json",
+   *   "categories": {
+   *     "correctness": "warn"
+   *   },
+   *   "rules": {
+   *     "eslint/no-unused-vars": "error"
+   *   }
+   * }
+   * ```
+   */
+  categories?: RuleCategories;
+  /**
+   * Environments enable and disable collections of global variables.
+   */
+  env?: OxlintEnv;
+  /**
+   * Configuration files that this configuration extends.
+   */
   extends?: OxlintConfig[];
+  /**
+   * Enabled or disabled specific global variables.
+   */
+  globals?: OxlintGlobals;
+  /**
+   * Globs to ignore during linting. These are resolved from the configuration file path.
+   */
+  ignorePatterns?: string[];
+  /**
+   * JS plugins, allows usage of ESLint plugins with Oxlint.
+   *
+   * Read more about JS plugins in
+   * [the docs](https://oxc.rs/docs/guide/usage/linter/js-plugins.html).
+   *
+   * Note: JS plugins are in alpha and not subject to semver.
+   */
+  jsPlugins?: null | ExternalPluginEntry[];
+  /**
+   * Oxlint config options.
+   */
+  options?: OxlintOptions;
+  /**
+   * Add, remove, or otherwise reconfigure rules for specific files or groups of files.
+   */
+  overrides?: OxlintOverrides;
+  /**
+   * Enabled built-in plugins for Oxlint.
+   * You can view the list of available plugins on
+   * [the website](https://oxc.rs/docs/guide/usage/linter/plugins.html#supported-plugins).
+   *
+   * NOTE: Setting the `plugins` field will overwrite the base set of plugins.
+   * The `plugins` array should reflect all of the plugins you want to use.
+   */
+  plugins?: LintPlugins;
+  /**
+   * Example
+   *
+   * `.oxlintrc.json`
+   *
+   * ```json
+   * {
+   *   "$schema": "./node_modules/oxlint/configuration_schema.json",
+   *   "rules": {
+   *     "eqeqeq": "warn",
+   *     "import/no-cycle": "error",
+   *     "prefer-const": [
+   *       "error",
+   *       {
+   *         "ignoreReadBeforeAssign": true
+   *       }
+   *     ]
+   *   }
+   * }
+   * ```
+   *
+   * See [Oxlint Rules](https://oxc.rs/docs/guide/usage/linter/rules.html) for the list of
+   * rules.
+   */
+  rules?: DummyRuleMap;
+  /**
+   * Plugin-specific configuration for both built-in and custom plugins.
+   * This includes settings for built-in plugins such as `react` and `jsdoc`
+   * as well as configuring settings for JS custom plugins loaded via `jsPlugins`.
+   */
+  settings?: OxlintPluginSettings;
 }
 /**
  * Define an Oxlint configuration with type inference.
@@ -6335,6 +6426,6 @@ interface OxlintConfig extends Oxlintrc {
  * @param config - Oxlint configuration
  * @returns Config unchanged
  */
-declare function defineConfig<T extends OxlintConfig>(config: T): T;
+declare function defineConfig<T extends OxlintConfig>(config: T & OxlintConfig): T;
 //#endregion
 export { type AllowWarnDeny, type DummyRule, type DummyRuleMap, type ExternalPluginEntry, type ExternalPluginsConfig, type OxlintConfig, type OxlintEnv, type OxlintGlobals, type OxlintOverride, type RuleCategories, defineConfig };
